src/tink/syntaxhub/FrontendContext.hx

```diff
@@ -98,26 +98,36 @@ class FrontendContext {
     }
     return ret;
   }
-  
+
   static function moduleForType(name:String) {
-    if (name.indexOf('__impl') != -1 || plugins.getData().length == 0) return None;
+    if (name.indexOf('__impl') != -1 || plugins.getData().length == 0) return;
     var pack = name.split('.');
-    var name = pack.pop();
-    var actual = pack.concat(['__impl', name]).join('.');
-    var exists = 
+    var tname = pack.pop();
+    var actual = pack.concat(['__impl', tname]).join('.');
+    cache[name] = {
+      pack: pack,
+      name: tname,
+      pos: Context.currentPos(),
+      fields: [],
+      kind: TDAlias(actual.asComplexType())
+    }
+    var exists =
       try {
-        Context.getModule(actual);
+        Context.getType(actual);
```

Comment:
I'm assuming that since the created type definition is a `TDAlias` to a particular type, that at least of the plugins will create it.

More importantly, this avoids a `onTypeNotFound` recursion where it would be very hard to handle correctly (without exposing a bunch of aliases to non existent implementations).